Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/LocationHandler.java

```diff
@@ -0,0 +1,153 @@
+package com.teamapricot.projectwalking;
+
+import android.Manifest;
+import android.location.LocationListener;
+import android.location.LocationManager;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A class for accessing location data.
+ */
+public class LocationHandler {
+    private final PermissionHandler permissionHandler;
+
+    private Position position = null;
+
+    private final LocationManager locationManager;
+    private LocationListener locationListener;
+
+    private final int updateInterval;
+
+    private final List<UpdateListener> updateListeners;
+
+    /**
+     * Constructor.
+     *
+     * @param activity The associated Activity
+     * @param interval Interval (in ms) between location updates
+     */
+    LocationHandler(AppCompatActivity activity, int interval) {
+        permissionHandler = new PermissionHandler(activity);
+        locationManager = (LocationManager) activity.getApplicationContext()
+                .getSystemService(AppCompatActivity.LOCATION_SERVICE);
+        updateListeners = new ArrayList<>();
+        updateInterval = interval;
+
+        if (permissionHandler.checkPermission(Manifest.permission.ACCESS_FINE_LOCATION)) {
+            initializeLocationHandler();
+        } else {
+            permissionHandler.requestPermissionAsync(Manifest.permission.ACCESS_FINE_LOCATION)
+                    .thenAccept(permissionGranted -> {
+                        if (permissionGranted) {
+                            initializeLocationHandler();
+                        } else {
+                            // TODO: Permission not granted, do something!
+                            System.out.println("permission not granted");
+                        }
+                    });
+        }
+    }
+
+    public Double getLatitude() {
+        if (position == null) {
+            return null;
+        }
+        return position.getLatitude();
+    }
+
+    public Double getLongitude() {
+        if (position == null) {
+            return null;
+        }
+        return position.getLongitude();
+    }
+
+    public Double getAccuracy() {
+        if (position == null) {
+            return null;
+        }
+        return position.getAccuracy();
+    }
```

Comment:
See comment about potentially adding a `Position` parameter to the `UpdateListener` interface. If you add the parameter, these methods will most likely not be useful when using this class, no?

Reply:
The user might not really care about when the position is updated. Registering a listener isn't required, so it should be possible to get the data without doing that.